api/anilist/src/main/kotlin/com/imashnake/animite/api/anilist/FlowExt.kt

```diff
@@ -0,0 +1,28 @@
+package com.imashnake.animite.api.anilist
+
+import com.apollographql.apollo3.api.ApolloResponse
+import com.apollographql.apollo3.api.Operation
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.mapLatest
+import java.io.IOException
+
+@OptIn(ExperimentalCoroutinesApi::class)
+fun <T: Operation.Data, R> Flow<ApolloResponse<T>>.asResult(transform: (T) -> R): Flow<Result<R>> = this
+    .mapLatest { response ->
+        if (response.data != null) {
+            Result.success(transform(response.dataAssertNoErrors))
+        } else if (response.hasErrors()) {
+            // TODO What kind of exception do we provide here?
+            throw IOException(response.errors!!.joinToString())
+        } else {
+            // No data, but also no exception. That's illegal
+            error("Unknown error occurred, no data or errors received.")
+        }
+    }
+    .catch { exception ->
+        emit(Result.failure(exception))
```

Comment:
We've essentially lost mapping to "human readable" error messages here. I think this makes sense, since this implementation uses `Result`, which is a lower level component that takes a Throwable rather than a string message. If we stick to this approach, Where should this translation be moved to?

ViewModel seems appropriate, but then we need some kind of centralized mapping.

Maybe the `Result.asResource` function should handle this?